src/Microsoft.DotNet.Wpf/src/PresentationBuildTasks/MS/Internal/MarkupCompiler/MarkupCompiler.cs

```diff
@@ -2596,7 +2596,20 @@ private void GenerateInitializeComponent(bool isApp)
 
             string uriPart = string.Empty;
 
-            string version = String.IsNullOrEmpty(AssemblyVersion) ? String.Empty : COMPONENT_DELIMITER + VER + AssemblyVersion;
+            bool hasWildcard = false;
+
+            // Attempt to parse out the AssemblyVersion if it exists.  This validates that we can either use an empty version string (wildcards exist)
+            // or we can utilize the passed in string (valid parse).
+            if (!string.IsNullOrEmpty(AssemblyVersion)
+                && !VersionHelper.TryParseAssemblyVersion(AssemblyVersion, allowWildcard: true, out Version _, out hasWildcard))
+            {
+                throw new Exception(SR.Get(SRID.InvalidAssemblyVersion, AssemblyVersion));
```

Comment:
Still investigating if throwing or using the OnError mechanism is more appropriate here.

With OnError we signal via an event and return, throwing *should* make its way up to a catch that does OnError at a higher level.  OnError is odd here since we may end up in a state where we continue to parse but haven't generated the file.  The throw is intended to short circuit that and stop immediately.

I am still testing this, so I marked it WIP and will investigate further.

Reply:
The error looks like this in VS (with a throw) and it comes up per XAML file.  The line numbers, etc, basically point to the first EndElement (this is where we generate `InitializeComponent` function).

I'm sort of on the fence about this, I might have to find a better way to plumb things all the way up beyond the markup compiler error paradigm.

![image](https://user-images.githubusercontent.com/20006987/76112773-b9cad680-5f97-11ea-8540-54e533b1cf1c.png)

Reply:
Also, FYI, `MC1005` is just the next number available in a series of errors that looks more akin to project configuration issues rather than actual parsing errors.  So I felt it better to group it there.

Reply:
MC1005 seems fine to me.

Reply:
It would be better if the errors mapped to the target of the task though.

Reply:
I can take a look at doing that, we'd have to have the plumbing to get the data back to the task output.  The MarkupCompiler is down a ways in the stack and EndElement is basically a parser event callback.

What we could do is throw a new exception type and, instead of trapping and converting to a MarkupException, allow it to bubble to the task and trap it specifically there, resulting in the output.

I'll think about it.

Reply:
I think that will work - just invent a new internal-only `Exception` type, and then when you catch it inside `Execute`...